### PR TITLE
Collect fee percentage for small numbers

### PR DESCRIPTION
**Pull request by carlosfebres**
## Summary

  - Fix precision loss in fee calculation by performing multiplication before division
  - Update test cases to match corrected calculations
  - Add comprehensive test coverage for edge cases with small values

 ## Details

  This PR addresses a precision issue in the fee calculation where integer division was happening before multiplication when calculating linear fees. In BigInt arithmetic, this can lead to significant
  precision loss, especially with smaller values.

  The solution rearranges the operation to multiply first, then divide:
  - Before: `(totalFulfill / unitSize) * unitFee`
  - After: `(totalFulfill * unitFee) / unitSize`

  This ensures that no precision is lost due to integer division truncation, providing more accurate fee calculations across all value ranges.

### CODE REVIEW

File: src/fee/tests/fee.service.spec.ts [hunk 1]
```diff
@@ -216,14 +216,47 @@ describe('FeeService', () => {
 
       it('should return the correct ask for less than $100', async () => {
         const ask = feeService.getAsk(1_000_000n, intent)
-        expect(ask).toBe(1_020_000n)
+        // 1_000_000n + 20_000n + (1_000_000n * 15_000n) / 100_000_000n = 1_000_000n + 20_000n + 150n = 1_020_150n
```
Comment (carlosfebres) [inline src/fee/tests/fee.service.spec.ts:219]: Note: The fee set here is 0.02$ as the base fee plus 0.015% of the total amount.

**Review by StoyanD — APPROVED** [review]
